src/Interpreters/ProcessList.cpp

```diff
@@ -89,6 +89,15 @@ ProcessList::EntryPtr ProcessList::insert(const String & query_, const IAST * as
                 throw Exception("Too many simultaneous queries. Maximum: " + toString(max_size), ErrorCodes::TOO_MANY_SIMULTANEOUS_QUERIES);
         }
 
+        {
```

Comment:
Please add a comment how does it differ from `max_size`.

Reply:
Tried my best.